.azure-pipelines/client.yml

```diff
@@ -214,3 +214,27 @@ jobs:
         inputs:
           mergeTestResults: true
           testRunTitle: '$(OSName) on Java $(JavaVersion)'
+
+        # Publish JaCoCo code coverage results from a build
+      - task: PublishCodeCoverageResults@1
+        displayName: 'Publish code coverage report'
+        inputs:
+            codeCoverageTool: 'JaCoCo'
+            summaryFileLocation: ./eng/jacoco-test-coverage/target/site/test-coverage/jacoco.xml
+            reportDirectory: ./eng/jacoco-test-coverage/target/site/test-coverage/
+            failIfCoverageEmpty: true
+
+        # Copy test coverage reports to respective OS and Java version directories
+      - task: CopyFiles@2
+        displayName: 'Copy test coverage reports to OS-JDKversion directory'
+        inputs:
+            sourceFolder: eng/jacoco-test-coverage/target/site/
+            targetFolder: test-coverage-$(OSName)-$(JavaVersion)
+
+        # Publish pipeline artifacts for respective OS and Java versions
+      - task: PublishPipelineArtifact@0
```

Comment:
You can add the following to only publish artifacts when a specific build matrix combination is present:

```yaml
condition: and(eq(variables['OsName'], 'Linux'), eq(variables['JavaVersion'], '1.11'))
```

Reply:
Since you only want one set of coverage reports to be pushed to blob storage, let's only save this one copy in the artifact named 'coverage'

Reply:
I would also update the comment above to clarify why we are only publishing the coverage report under this condition